src/utils/routes.js

```diff
@@ -85,6 +85,10 @@ function getLoginUrl(redirect = null) {
   return redirect ? urlUtils.createRedirectUrl(loginPath, redirect) : loginPath;
 }
 
+function getLoginUrlForCurrentUrl(currentUrl) {
```

Comment:
I think we don't need this method, we could put the ternary operator in the `getLoginUrl`. For what this method itself knows, the `currentUrl` could be "any" url, which in the `getLoginUrl` method is called `redirect`.

Reply:
What I would do would be to make a single helper method `getLoginUrl` 

```
function getLoginUrl({ currentUrl, ensuredRedirectUrl } = {}) {
  if (ensuredRedirectUrl) {
    return urlUtils.createRedirectUrl(loginPath, ensuredRedirectUrl);
  }

  return currentUrl === homePath ? currentUrl : urlUtils.createRedirectUrl(loginPath, currentUrl);
}
```

To me this is easier to understand (as there is but one method to use) and it separates clearly the single case in which we need to `ensuredRedirectUrl`, which is when the `error-controller` calls this method.

Reply:
I reworked both functions into one as discussed. It is now used in the same manner everywhere.

Reply:
Nice, clear and simple!